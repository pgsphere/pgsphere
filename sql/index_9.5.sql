-- test spoint3 operator class with and without index-only scan

SET enable_seqscan = OFF;
SET enable_bitmapscan = OFF;
SET enable_indexonlyscan = ON;

EXPLAIN (COSTS OFF) SELECT count(*) FROM spheretmp1b WHERE p <@ scircle '<(1,1),0.3>';
                    SELECT count(*) FROM spheretmp1b WHERE p <@ scircle '<(1,1),0.3>';
EXPLAIN (COSTS OFF) SELECT count(*) FROM spheretmp1b WHERE p = spoint '(3.09 , 1.25)';
                    SELECT count(*) FROM spheretmp1b WHERE p = spoint '(3.09 , 1.25)';

SET enable_bitmapscan = ON;
SET enable_indexonlyscan = OFF;

EXPLAIN (COSTS OFF) SELECT count(*) FROM spheretmp1b WHERE p <@ scircle '<(1,1),0.3>';
                    SELECT count(*) FROM spheretmp1b WHERE p <@ scircle '<(1,1),0.3>';
EXPLAIN (COSTS OFF) SELECT count(*) FROM spheretmp1b WHERE p = spoint '(3.09 , 1.25)';
                    SELECT count(*) FROM spheretmp1b WHERE p = spoint '(3.09 , 1.25)';
