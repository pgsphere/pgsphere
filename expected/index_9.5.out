-- test spoint3 operator class with and without index-only scan
SET enable_seqscan = OFF;
SET enable_bitmapscan = OFF;
SET enable_indexonlyscan = ON;
EXPLAIN (COSTS OFF) SELECT count(*) FROM spheretmp1b WHERE p <@ scircle '<(1,1),0.3>';
                       QUERY PLAN                       
--------------------------------------------------------
 Aggregate
   ->  Index Only Scan using spoint3_idx on spheretmp1b
         Index Cond: (p <@ '<(1 , 1) , 0.3>'::scircle)
(3 rows)

                    SELECT count(*) FROM spheretmp1b WHERE p <@ scircle '<(1,1),0.3>';
 count 
-------
    32
(1 row)

EXPLAIN (COSTS OFF) SELECT count(*) FROM spheretmp1b WHERE p = spoint '(3.09 , 1.25)';
                       QUERY PLAN                       
--------------------------------------------------------
 Aggregate
   ->  Index Only Scan using spoint3_idx on spheretmp1b
         Index Cond: (p = '(3.09 , 1.25)'::spoint)
(3 rows)

                    SELECT count(*) FROM spheretmp1b WHERE p = spoint '(3.09 , 1.25)';
 count 
-------
     4
(1 row)

SET enable_bitmapscan = ON;
SET enable_indexonlyscan = OFF;
EXPLAIN (COSTS OFF) SELECT count(*) FROM spheretmp1b WHERE p <@ scircle '<(1,1),0.3>';
                      QUERY PLAN                       
-------------------------------------------------------
 Aggregate
   ->  Index Scan using spoint3_idx on spheretmp1b
         Index Cond: (p <@ '<(1 , 1) , 0.3>'::scircle)
(3 rows)

                    SELECT count(*) FROM spheretmp1b WHERE p <@ scircle '<(1,1),0.3>';
 count 
-------
    32
(1 row)

EXPLAIN (COSTS OFF) SELECT count(*) FROM spheretmp1b WHERE p = spoint '(3.09 , 1.25)';
                    QUERY PLAN                     
---------------------------------------------------
 Aggregate
   ->  Index Scan using spoint3_idx on spheretmp1b
         Index Cond: (p = '(3.09 , 1.25)'::spoint)
(3 rows)

                    SELECT count(*) FROM spheretmp1b WHERE p = spoint '(3.09 , 1.25)';
 count 
-------
     4
(1 row)

